src/main/java/net/fabricmc/loader/metadata/ModMetadataV1.java

```diff
@@ -572,4 +564,30 @@ public LicenseEntry deserialize(JsonElement json, Type typeOfT, JsonDeserializat
 			}
 		}
 	}
+
+	public static class CustomValueContainer {
+		private Map<String, CustomValue> customValues = new HashMap<>();
+
+		private Map<String, CustomValue> getCustomValues() {
+			return Collections.unmodifiableMap(this.customValues);
```

Comment:
maybe store the unmodifiable map wrapper in a field so you don't create one for each call

Reply:
Well I'd say player should comment on that, we already end up recreating some lists and maps such as getting entrypoint containers here:
https://github.com/FabricMC/fabric-loader/blob/master/src/main/java/net/fabricmc/loader/EntrypointStorage.java#L177